### PR TITLE
Chore: remove refs to deprecated io/ioutil 

### PR DESCRIPTION
**Pull request by kacperswis**
FIXES: #[2806](https://github.com/GoogleContainerTools/kaniko/issues/2806)

**Description**

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [x] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit tests and or integration tests added.

### CODE REVIEW

**Comment by JeromeJu** [comment]
Thanks for the fix @kacperswis would you mind squashing the commits as some of the seems to be "WIP" steps?

**Comment by kacperswis** [comment]
Do you mean to make another PR and "merge" commits? @JeromeJu

**Comment by JeromeJu** [comment]
> Do you mean to make another PR and "merge" commits? @JeromeJu

I was referring to the 3 commits in this PR that all come together for the purpose of removing deprecated io/ioutil, where they can [be squashed using command `git rebase -i <previous commit sha>`](https://gist.github.com/lpranam/4ae996b0a4bc37448dc80356efbca7fa):
```
[chore: remove refs to deprecated io/ioutil](https://github.com/GoogleContainerTools/kaniko/pull/2807/commits/f17da1f7ae13a45f802fa974f60de61466992e10)

[Merge branch 'main' into chore-remove-refs-to-deprecated-io/ioutil-](https://github.com/GoogleContainerTools/kaniko/pull/2807/commits/e51b7f48004118c20d2d7555fe2e6ea54079b16f)

[chore: repair of erroneously resolved conflictt](https://github.com/GoogleContainerTools/kaniko/pull/2807/commits/3a1e8e0c2557cee18273edc463b3d8d7d71b3ccb)

```

It'd help other contributors to better understand the PR context if we have a commit dedicated to tackle one specific issue.

**Review by aaron-prindle — APPROVED** [review]
I believe we can squash and merge this on our end directly, going to try that now.  I think there may be some errors w/ this PR from test changes as I am going through this

File: pkg/commands/copy_test.go [hunk 1]
```diff
@@ -451,13 +450,14 @@ func TestCopyCommand_ExecuteCommand_Extended(t *testing.T) {
...
-		actual, err := ioutil.ReadDir(filepath.Join(testDir, "dest"))
+		actual, err := os.ReadDir(filepath.Join(testDir, "dest"))
 		if err != nil {
 			t.Fatal(err)
 		}
 		for i, f := range actual {
+			fi, _ := expected[i].Info()
 			testutil.CheckDeepEqual(t, expected[i].Name(), f.Name())
-			testutil.CheckDeepEqual(t, expected[i].Mode(), f.Mode())
+			testutil.CheckDeepEqual(t, fi.Mode(), fi.Mode())
```
Comment (aaron-prindle) [inline pkg/commands/copy_test.go:460]: This change is not related to the PR goals and actually breaks the test IIUC (comparing `fi.Mode()` w/ `fi.Mode()` now).  This change should likely be undone

**Review by aaron-prindle — CHANGES_REQUESTED** [review]

File: pkg/commands/copy_test.go [hunk 2]
```diff
@@ -565,14 +565,15 @@ func TestCopyCommand_ExecuteCommand_Extended(t *testing.T) {
...
-		files, err := ioutil.ReadDir(filepath.Join(testDir, "dest"))
+		files, err := os.ReadDir(filepath.Join(testDir, "dest"))
 		if err != nil {
 			t.Fatal(err)
 		}
 		// bam.txt and sym.link should be present
 		testutil.CheckDeepEqual(t, 1, len(files))
 		testutil.CheckDeepEqual(t, files[0].Name(), "sym.link")
-		testutil.CheckDeepEqual(t, true, files[0].Mode()&os.ModeSymlink != 0)
+		fi, _ := files[0].Info()
```
Comment (aaron-prindle) [inline pkg/commands/copy_test.go:575]: nit: can you just inline this vs making the `fi` var?

File: pkg/commands/copy_test.go [hunk 3]
```diff
@@ -610,13 +611,14 @@ func TestCopyCommand_ExecuteCommand_Extended(t *testing.T) {
...
 		// Check if "dest" dir exists with link dead.link
-		files, err := ioutil.ReadDir(filepath.Join(testDir, "dest"))
+		files, err := os.ReadDir(filepath.Join(testDir, "dest"))
 		if err != nil {
 			t.Fatal(err)
 		}
 		testutil.CheckDeepEqual(t, 1, len(files))
 		testutil.CheckDeepEqual(t, files[0].Name(), "dead.link")
-		testutil.CheckDeepEqual(t, true, files[0].Mode()&os.ModeSymlink != 0)
+		fi, _ := files[0].Info()
```
Comment (aaron-prindle) [inline pkg/commands/copy_test.go:620]: nit: can you just inline this vs making the `fi` var?

File: pkg/commands/copy_test.go [hunk 4]
```diff
@@ -698,14 +700,16 @@ func TestCopyCommand_ExecuteCommand_Extended(t *testing.T) {
...
 		// Check if "dest" dir exists contents of srcDir and an extra zSym.link created
 		// in this test
-		actual, err := ioutil.ReadDir(filepath.Join(testDir, "dest"))
+		actual, err := os.ReadDir(filepath.Join(testDir, "dest"))
 		if err != nil {
 			t.Fatal(err)
 		}
 		testutil.CheckDeepEqual(t, 4, len(actual))
 		for i, f := range expected {
+			fi, _ := f.Info()
```
Comment (aaron-prindle) [inline pkg/commands/copy_test.go:709]: nit: can you just inline this vs making the `fi` and `afi` var?

File: pkg/commands/copy_test.go [hunk 5]
```diff
@@ -784,13 +788,15 @@ func TestCopyCommand_ExecuteCommand_Extended(t *testing.T) {
 		err = cmd.ExecuteCommand(cfg, dockerfile.NewBuildArgs([]string{}))
 		testutil.CheckNoError(t, err)
 		// Check if "linkdest" dir exists with contents of srcDir
-		actual, err := ioutil.ReadDir(filepath.Join(testDir, "linkDest"))
+		actual, err := os.ReadDir(filepath.Join(testDir, "linkDest"))
 		if err != nil {
 			t.Fatal(err)
 		}
 		for i, f := range expected {
+			fi, _ := f.Info()
```
Comment (aaron-prindle) [inline pkg/commands/copy_test.go:796]: nit: can you just inline this vs making the `fi` and `afi` var?

File: pkg/commands/copy_test.go [hunk 6]
```diff
@@ -878,14 +884,14 @@ func TestCopyCommand_ExecuteCommand_Extended(t *testing.T) {
...
-		actual, err := ioutil.ReadDir(filepath.Join(testDir))
+		actual, err := os.ReadDir(filepath.Join(testDir))
 		if err != nil {
 			t.Fatal(err)
 		}
 
 		testutil.CheckDeepEqual(t, "bam.txt", actual[0].Name())
-
-		if stat, ok := actual[0].Sys().(*syscall.Stat_t); ok {
+		fi, _ := actual[0].Info()
```
Comment (aaron-prindle) [inline pkg/commands/copy_test.go:893]: nit: can you just inline this vs making the `fi` var?

**Review by aaron-prindle — CHANGES_REQUESTED** [review]

**Comment by aaron-prindle** [comment]
https://github.com/GoogleContainerTools/kaniko/issues/2806 is fixed now, closing

**Pull request closed without merging**